File: app/src/main/java/com/nanodegree/popularmovies/service/component/ServiceComponent.java
package com.nanodegree.popularmovies.service.component;

import android.content.SharedPreferences;

import com.nanodegree.popularmovies.AppModule;
import com.nanodegree.popularmovies.service.module.ServiceModule;

import javax.inject.Singleton;

import dagger.Component;
import retrofit2.Retrofit;

/**
 * Created by luan_ on 31/05/2017.
 */

@Singleton
@Component(modules = {AppModule.class, ServiceModule.class})
public interface ServiceComponent {
    Retrofit retrofit();
    SharedPreferences sharedPreferences();
}

File: app/src/main/java/com/nanodegree/popularmovies/service/component/ServiceComponent.kt
package com.nanodegree.popularmovies.service.component

import com.nanodegree.popularmovies.service.module.ServiceModule
import dagger.Component
import retrofit2.Retrofit
import javax.inject.Singleton


/**
 * Created by luanfernandes on 31/08/17.
 */
@Singleton
@Component(modules = arrayOf(ServiceModule::class))
interface ServiceComponent {
    fun retrofit(): Retrofit
}
